### ISSUE TITLE
InsertFileFromBase64 is not working in Word Online

### ISSUE THREAD

**Issue by nickdekesel** [OfficeDev/office-js#3573]
# Provide required information needed to triage your issue
When using the insertFileFromBase64 in Word online the document does not update with the inserted content. Only after you refresh the page the content is actually shown. In Word on PC desktop and Mac the issue is not present.

## Your Environment

* Platform [PC desktop, Mac, iOS, Office on the web]: Office on the web
* Host [Excel, Word, PowerPoint, etc.]: Word
* Office version number: 16.0.16803.41000
* Operating System: Windows
* Browser (if using Office on the web): Chrome
 
## Expected behavior

Inserted content should be visible in the open document

## Current behavior

There is a pop up with a spinner saying 'Waiting...'. After some time this goes away and the document is still empty.

## Steps to reproduce

1. Create a script in script lab that uses insertFileFromBase64 with a very basic base64 document
2. Insert via script lab
3. See that it does not update the content

## Link to live example(s)

https://gist.github.com/nickdekesel/c151d60068f20c07a940fd96f159a71e

## Context

We are trying to insert content based on base64 files in our product which we fail to do in Word online as of now.

**Comment by xiaoyuMS** [comment]
Hi @nickdekesel , we have another thread talking about this API https://github.com/OfficeDev/office-js/issues/3040. It is a known limitation on this API.

**Comment by nickdekesel** [comment]
Hi @xiaoyuMS, thanks for the reply. For us it is a new issue facing our customers making it not able to use our product properly. Is there a way to get this bug prioritized?

**Comment by xiaoyuMS** [comment]
@nickdekesel If possible, could you share the business impact of this issue?

**Comment by ZYUN-MSFT** [comment]
Hi @nickdekesel ,

Please provided the business impact of your scenario (or your Addin name in the office store) if it's possible.\
That woulbe be helpful a lot for us to re-priority the work-item.

Thanks.

**Comment by nickdekesel** [comment]
@ZYUN-MSFT The add-in in question is Templafy but we are not in the store, we distribute the add-in to organizations to be managed on admin level.

**Comment by ZYUN-MSFT** [comment]
@nickdekesel Thanks for your information. We will feed back this to our engineering team.

**Comment by JBoman32768** [comment]
I can confirm this issue also affects the insertOoxml function as well.

This impacts the users of our Project Online Sync for Word product ( https://docs.altus.pro/solutions/POSW/index.html )

We store pieces of template as OOXML and then insert them.

The error is the same as the OP.  Popup "Waiting..." and then eventually: 

```RichApi.Error: Can't find the inserted content. Please refresh the page for the result.```

Word Online 16.0.16901.41002
Document host is SharePoint.

**Comment by xiaoyu03** [comment]
@JBoman32768 , insertOoxml() shares the same infrastructure as insertFileFromBase64() and it is a known API issue on Word Web.   This API is not recommended on Word Web now. 

How often does it fail to find the content？

**Comment by JBoman32768** [comment]
> insertOoxml() shares the same infrastructure as insertFileFromBase64() and it is a known API issue on Word Web. This API is not recommended on Word Web now.

I understand what you mean, but there doesn't appear to be a workaround and more broadly I think API's should be either supported, or not supported.

> How often does it fail to find the content？

Every time I use it, I get a popup that says "Inserting..." and then another that says "Waiting..." and then it eventually fails.  It does end up inserting the content, so after refreshing the page the content control is in there - but its not a good workflow for end-users.

**Comment by JPfahl** [comment]
This is a show stopper and a regression, how can this not be a priority to fix for the NEW method for building addins?

**Comment by JBoman32768** [comment]
This also appears to be logged as https://github.com/OfficeDev/office-js/issues/3216

**Comment by xiaoyuMS** [comment]
@JBoman32768  @JPfahl Thanks for reaching us, we are fixing this. I replied to another thread #3040

**Comment by xiaoyuMS** [comment]
Hi @JBoman32768 @JPfahl, the fix has been deployed to production, please have a try. The full deployment may take another 1-2 days.

**Comment by JBoman32768** [comment]
> Hi @JBoman32768 @JPfahl, the fix has been deployed to production, please have a try. The full deployment may take another 1-2 days.

Hey there! I tested it today and it does appear to be intermittently fixed.  Will retest next week - but big thanks for attending to this issue 👍

**Comment by xiaoyuMS** [comment]
@JBoman32768 Thanks for catching this, we found a case that we missed as well. There is a bug that was created #VSO8511787. In some cases, the problem arises when we try to insert a single paragraph range.

**Comment by xiaoyuMS** [comment]
Hi @JBoman32768, the fix has been deployed to production, please have a try to make sure your use cases can work.

**Comment by JPfahl** [comment]
I just tested this and I get the following error:

name: 'RichApi.Error', code: 'NotAllowed', traceMessages: {}, innerError: null, debugInfo: {code: 'NotAllowed', message: 'The action isn't supported by Word in a browser. Check the OfficeExtension.Error.debugInfo for more information.', errorLocation: 'Document.insertFileFromBase64', statement: 'var insertFileFromBase64 = root.insertFileFromBase64(...);', ...}, httpStatusCode: 500, data: undefined

**Comment by xiaoyuMS** [comment]
@JPfahl I can't repro it on my side, could you provide the repro step?  Or you can try body.insertFileFromBase64().

**Comment by JPfahl** [comment]
The following code causes the error: Uncaught (in promise) RichApi.Error: The action isn't supported by Word in a browser.
    This works fine in Word in Windows.

    Word.run(async (context: Word.RequestContext) =>
    {
        context.document.body.load();
        await context.sync();
        const xml = context.document.body.getOoxml();
        await context.sync();
        
        context.document.insertFileFromBase64(
            btoa(unescape(encodeURIComponent(xml.value))),
            Word.InsertLocation.replace);
        await context.sync();
    });

**Comment by xiaoyu03** [comment]
@JPfahl It looks like the Document.insertFileFromBase64() isn't supported in your Word Web version and is not a bug. Could you provide your build number?
![image](https://github.com/OfficeDev/office-js/assets/67049434/0142e772-960e-4bd5-9276-a340555e5234)

By the way, I think the insertFileFromBase64 on the Body object should work since it was released a long time ago.

**Comment by JPfahl** [comment]
Replacing the body will not work for me because I need to update the entire document.

![image](https://github.com/OfficeDev/office-js/assets/101600436/99af7d34-210a-4ee3-818e-7ff7d60751e4)

**Comment by Yuchao9145** [comment]
This issue still happening and seems to only happens to certain docx file content.

```js
await Word.run(async (context: any) => {
          // Use the Base64-encoded string representation of the selected .docx file.
          context.document.insertFileFromBase64(encodedContent, 'Replace', {
            importTheme: true,
            importStyles: true,
            importParagraphSpacing: true,
            importPageColor: true,
            importChangeTrackingMode: true,
            importCustomProperties: true,
            importCustomXmlParts: true,
            importDifferentOddEvenPages: true
          });
          try {
            await context.sync();
```

In my end I'm constantly getting this error with this file

[1 (1).docx](https://github.com/OfficeDev/office-js/files/15004112/1.1.docx)


![image](https://github.com/OfficeDev/office-js/assets/159736769/0bcf6c4f-0af3-4676-9b4f-866d103b9124)

**Comment by xiruatms** [comment]
@Yuchao9145 , it seems your case is a different one with specific documents, will you please create another issue to track your case? Thanks~

@JPfahl , how about your issue now? are you still encountering that? Seem to us a different one from this. If you can still repro it, please create another issue to track. Thanks.
